app/models/spree/payment_method/klarna_credit.rb

```diff
@@ -93,12 +91,6 @@ def try_void(payment)
       def spree_order(options)
         Spree::Order.find_by(number: options[:order_id].split("-").first)
       end
-
-      def tokenization_flow
```

Comment:
You are right, now the customer token is saved on the payment source but a customer token shouldn't be created only if the user is logged in? If you enable the tokenization flow from the admin, the extension will use the intent TOKENIZE but the user is not logged in.

Reply:
What is the problem with it? even when the user is not logged in, Klarna will generate the authorization token and the extension will convert it to the customer token, but the user won't be able to re-use it for other payments. Does it make sense for you?

Reply:
@vassalloandrea considering your thought I have enabled limited tokenization only for not guest orders.